serviceGroups resourceType

ramoka178 opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [X] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [X] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [X] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

ramoka178 committed:
Base commit: Copying from 2023-04-01 to 2024-02-01-preview

ramoka178 committed:
Update API version for existing Swaggers/Examples

ramoka178 committed:
Update for service groups APIs

ramoka178 committed:
keeping title consistent

ramoka178 committed:
move to v5

ramoka178 committed:
fix x-ms-example

ramoka178 commented:
![image](https://github.com/user-attachments/assets/aaa8565a-4de9-4680-8a28-cefc979a42ae)

Approval from PAS . When we got the approval, the resourceType was named as Applications and so the email says that.
It got later renamed to ServiceGroups, which the PR has today.

gary-x-li commented on specification/managementgroups/resource-manager/Microsoft.Management/preview/2024-02-01-preview/serviceGroups.json:
```diff
@@ -0,0 +1,312 @@
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
```
>"final-state-via": "azure-async-operation" [](http://example.com/codeflow?start=10&length=42)

Change to "Location" so it's consistent with the header above?

gary-x-li commented on specification/managementgroups/resource-manager/Microsoft.Management/preview/2024-02-01-preview/serviceGroups.json:
```diff
@@ -0,0 +1,312 @@
+        }
+      }
+    },
+    "/providers/Microsoft.Management/serviceGroups/{serviceGroupName}/listAncestors": {
```
>listAncestors [](http://example.com/codeflow?start=70&length=13)

Why using POST instead of GET?

ramoka178 replied:
serviceGroups is the resourceType. This is an action of listing the ancestors of a given serviceGroup. 

This serves a collection response and a collection GET API path should end with resourceType, that matches the response items, which is not the case we have. 

so, opted for POST.

gary-x-li commented:
Please also fix the LintDiff error.

ramoka178 committed:
fix async final state

ramoka178 commented:
> Please also fix the LintDiff error.

The lintdiff in this PR is due to Tenant Level APIs. We are a tenant level RP and only have tenant level resources. 
Added related suppression to it. 
Could you please check and provide suppression review too ?

gary-x-li commented:
> > Please also fix the LintDiff error.
> 
> The lintdiff in this PR is due to Tenant Level APIs. We are a tenant level RP and only have tenant level resources. Added related suppression to it. Could you please check and provide suppression review too ?

What's suppressed in readme.md is a different one (`TopLevelResourcesListBySubscription`). Can you add `TenantLevelAPIsNotAllowed` as well?

ramoka178 committed:
fix suppression

ramoka178 commented:
> > > Please also fix the LintDiff error.
> > 
> > 
> > The lintdiff in this PR is due to Tenant Level APIs. We are a tenant level RP and only have tenant level resources. Added related suppression to it. Could you please check and provide suppression review too ?
> 
> What's suppressed in readme.md is a different one (`TopLevelResourcesListBySubscription`). Can you add `TenantLevelAPIsNotAllowed` as well?

My bad. Updated to have TenantLevelAPIsNotAllowed as that is being flagged in LintDiff

gary-x-li commented:
> > > > Please also fix the LintDiff error.
> > > 
> > > 
> > > The lintdiff in this PR is due to Tenant Level APIs. We are a tenant level RP and only have tenant level resources. Added related suppression to it. Could you please check and provide suppression review too ?
> > 
> > 
> > What's suppressed in readme.md is a different one (`TopLevelResourcesListBySubscription`). Can you add `TenantLevelAPIsNotAllowed` as well?
> 
> My bad. Updated to have TenantLevelAPIsNotAllowed as that is being flagged in LintDiff

The check is still failing. You'll probably need to adjust the `where` clause.

ramoka178 committed:
fix suppression

ramoka178 commented:
> > > > > Please also fix the LintDiff error.
> > > > 
> > > > 
> > > > The lintdiff in this PR is due to Tenant Level APIs. We are a tenant level RP and only have tenant level resources. Added related suppression to it. Could you please check and provide suppression review too ?
> > > 
> > > 
> > > What's suppressed in readme.md is a different one (`TopLevelResourcesListBySubscription`). Can you add `TenantLevelAPIsNotAllowed` as well?
> > 
> > 
> > My bad. Updated to have TenantLevelAPIsNotAllowed as that is being flagged in LintDiff
> 
> The check is still failing. You'll probably need to adjust the `where` clause.

thanks for the pointer. Updated and now the check passes.

msyyc committed:
Create sdk-suppressions.yaml

kazrael2119 committed:
Update sdk-suppressions.yaml

ramoka178 commented:
Created a release branch "release-managementGroups20240201preview" as PR to main is forcing to add "PublishToCustomers" label. This code is yet to be public preview.

ramoka178 committed:
Merge branch 'release-managementGroups20240201preview' into ramoka/serviceGroups
